ibm_watson/visual_recognition_v4.py

```diff
@@ -0,0 +1,2480 @@
+                             disable_ssl_verification=disable_ssl_verification)
+        self.version = version
+
+    #########################
+    # Analysis
+    #########################
+
+    def analyze(self,
+                collection_ids,
+                features,
+                *,
+                images_file=None,
+                image_url=None,
+                threshold=None,
+                **kwargs):
+        """
+        Analyze images.
+
+        Analyze images by URL, by file, or both against your own collection. Make sure
+        that **training_status.objects.ready** is `true` for the feature before you use a
+        collection to analyze images.
+        Encode the image and .zip file names in UTF-8 if they contain non-ASCII
+        characters. The service assumes UTF-8 encoding if it encounters non-ASCII
+        characters.
+
+        :param list[str] collection_ids: The IDs of the collections to analyze.
+        :param list[str] features: The features to analyze.
+        :param list[FileWithMetadata] images_file: (optional) An array of image
+               files (.jpg or .png) or .zip files with images.
+               - Include a maximum of 20 images in a request.
+               - Limit the .zip file to 100 MB.
+               - Limit each image file to 10 MB.
+               You can also include an image with the **image_url** parameter.
+        :param list[str] image_url: (optional) An array of URLs of image files
+               (.jpg or .png).
+               - Include a maximum of 20 images in a request.
+               - Limit each image file to 10 MB.
+               - Minimum width and height is 30 pixels, but the service tends to perform
+               better with images that are at least 300 x 300 pixels. Maximum is 5400
+               pixels for either height or width.
+               You can also include images with the **images_file** parameter.
+        :param float threshold: (optional) The minimum score a feature must have to
+               be returned.
+        :param dict headers: A `dict` containing the request headers
+        :return: A `DetailedResponse` containing the result, headers and HTTP status code.
+        :rtype: DetailedResponse
+        """
+
+        if collection_ids is None:
+            raise ValueError('collection_ids must be provided')
+        if features is None:
+            raise ValueError('features must be provided')
+
+        headers = {}
+        if 'headers' in kwargs:
+            headers.update(kwargs.get('headers'))
+        sdk_headers = get_sdk_headers('watson_vision_combined', 'V4', 'analyze')
+        headers.update(sdk_headers)
+
+        params = {'version': self.version}
+
+        form_data = []
+        for item in collection_ids:
+            form_data.append(('collection_ids', (None, item, 'text/plain')))
+        for item in features:
+            form_data.append(('features', (None, item, 'text/plain')))
```

Comment:
Do we need to convert collection_ids and features into csv here? I think they are lists. The service expects a single form part with a csv